Source/WebCore/xml/XPathParser.cpp

```diff
@@ -47,16 +47,31 @@ namespace XPath {
 
 struct Parser::Token {
     int type;
-    String string;
-    Step::Axis axis;
-    NumericOp::Opcode numericOpcode;
-    EqTestOp::Opcode equalityTestOpcode;
-
-    Token(int type) : type(type) { }
-    Token(int type, const String& string) : type(type), string(string) { }
-    Token(int type, Step::Axis axis) : type(type), axis(axis) { }
-    Token(int type, NumericOp::Opcode opcode) : type(type), numericOpcode(opcode) { }
-    Token(int type, EqTestOp::Opcode opcode) : type(type), equalityTestOpcode(opcode) { }
+    using TokenValue = std::variant<String, Step::Axis, NumericOp::Opcode, EqTestOp::Opcode>;
+    TokenValue value;
+
+    Token() = delete;
+
+    Token(int type)
+        : type(type)
+    { }
+    Token(int type, const String& string)
+        : type(type), value(string)
+    { }
+    Token(int type, Step::Axis axis)
+        : type(type), value(axis)
+    { }
+    Token(int type, NumericOp::Opcode opcode)
+        : type(type), value(opcode)
+    { }
+    Token(int type, EqTestOp::Opcode opcode)
+        : type(type), value(opcode)
+    { }
+
+    String& string() { return std::get<String>(value); }
+    Step::Axis axis() { return std::get<Step::Axis>(value); }
```

Comment:
function could be const.

Reply:
You know you can highlight multiple lines by click-and-drag so you only have to write a single comment?  :)

Reply:
Ah! Today I learn :)